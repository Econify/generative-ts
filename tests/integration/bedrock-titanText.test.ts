import {
  AmazonTitanTextApi,
  createAwsBedrockModelProvider,
} from "@packages/core";

test("Bedrock - Amazon TitanText", async () => {
  // arrange
  const titanText = createAwsBedrockModelProvider({
    api: AmazonTitanTextApi,
    modelId: "amazon.titan-text-express-v1",
  });

  // act
  const response = await titanText.sendRequest({
    prompt: "Brief history of NY Mets:",
    maxTokenCount: 100,
  });

  // assert
  expect(response).toMatchApiSnapshot();
});
