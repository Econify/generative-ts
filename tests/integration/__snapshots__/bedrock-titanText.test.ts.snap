// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bedrock - Amazon TitanText 1`] = `
{
  "inputTextTokenCount": "number",
  "results": [
    {
      "completionReason": "string",
      "outputText": "string",
      "tokenCount": "number",
    },
  ],
}
`;
